src/controllers/hotels.controller.js

```diff
@@ -105,6 +105,51 @@ class Hotel {
 
     return Responses.handleSuccess(201, 'Room added successfully', res, room);
   }
+
+  /**
+   * Get all the hotels with likes
+   * @param {*} req
+   * @param {*} res
+   * @returns {Object} All hotels
+   */
+  async getAllHotels(req, res) {
+    const hotel = await hotelService.getHotels();
+    return Responses.handleSuccess(200, 'Hotels retrieved successfully', res, hotel);
+  }
+
+  /**
+   * Get hotel with likes and rooms
+   * @param {*} req
+   * @param {*} res
+   * @returns {Object} hotel
+   */
+  async getHotel(req, res) {
+    const { hotelId } = req.params;
+    const hotel = await hotelService.getHotelById(hotelId);
+    return Responses.handleSuccess(200, 'hotel retrieved successfully', res, hotel);
+  }
+
+  /**
+   * Like or unlike accomodation
+   * @param {*} req
+   * @param {*} res
+   * @returns {Object} response
+   */
+  async like(req, res) {
+    const { hotelId } = req.params;
+    const { userId } = res.locals.user;
+
+    const hasLiked = await hotelService.hasLiked(hotelId, userId);
+    if (hasLiked) {
+      await hotelService.unLike(hotelId, userId);
+    } else {
+      await hotelService.like(hotelId, userId);
+    }
+
+    const hotel = await hotelService.getHotelById(hotelId);
+
+    return Responses.handleSuccess(200, 'Liked successfully', res, hotel);
```

Comment:
Looks like we are returning the same message `Liked successfully` even when an `unlike` operation has been performed.

Reply:
Done, thanks